picamera2/picamera2.py

```diff
@@ -964,16 +1036,13 @@ def stop_(self, request=None) -> None:
             self.stop_count += 1
             self.camera.stop()
 
-            # Temporary hack to flush Requests from the event queue.
+            # Flush Requests from the event queue.
             # This is needed to prevent old completed Requests from showing
             # up when the camera is started the next time.
-            sel = selectors.DefaultSelector()
-            sel.register(self.camera_manager.event_fd, selectors.EVENT_READ)
-            events = sel.select(0)
-            if events:
-                self.camera_manager.get_ready_requests()
-
+            self._cm.handle_request(self.camera_idx)
             self.started = False
+            with self._requestslock:
```

Comment:
I guess that taking the lock is probably optional here, given that we've ensured the camera can't complete any more requests and we've read out any that were still pending. Still, it's harmless!